packages/app-runtime/test/lib/MockUIBridge.matchers.ts

```diff
@@ -0,0 +1,122 @@
+import { MockUIBridge } from "./MockUIBridge";
+
+expect.extend({
+    showDeviceOnboardingCalled(mockUIBridge: unknown, deviceId: string) {
+        if (!(mockUIBridge instanceof MockUIBridge)) {
+            throw new Error("This method can only be used with expect(MockUIBridge).");
+        }
+
+        const calls = mockUIBridge.calls.filter((x) => x.method === "showDeviceOnboarding");
+        if (calls.length === 0) {
+            return { pass: false, message: () => "The method showDeviceOnboarding was not called." };
+        }
+
+        const matchingCalls = calls.filter((x) => x.deviceOnboardingInfo.id === deviceId);
+        if (matchingCalls.length === 0) {
+            return {
+                pass: false,
+                message: () =>
+                    `The method showDeviceOnboarding was called, but not with the specified device id '${deviceId}', instead with ids '${calls.map((e) => e.deviceOnboardingInfo.id).join(", ")}}'.`
+            };
+        }
+
+        return { pass: true, message: () => "" };
+    },
+    showDeviceOnboardingNotCalled(mockUIBridge: unknown) {
+        if (!(mockUIBridge instanceof MockUIBridge)) {
+            throw new Error("This method can only be used with expect(MockUIBridge).");
+        }
+
+        const calls = mockUIBridge.calls.filter((x) => x.method === "showDeviceOnboarding");
+        if (calls.length > 0) {
+            return { pass: false, message: () => "The method showDeviceOnboarding was called." };
+        }
+
+        return { pass: true, message: () => "" };
+    },
+    requestAccountSelectionCalled(mockUIBridge: unknown, possibleAccountsLength: number) {
+        if (!(mockUIBridge instanceof MockUIBridge)) {
+            throw new Error("This method can only be used with expect(MockUIBridge).");
+        }
+
+        const calls = mockUIBridge.calls.filter((x) => x.method === "requestAccountSelection");
+        if (calls.length === 0) {
+            return { pass: false, message: () => "The method requestAccountSelection was not called." };
+        }
+
+        const matchingCalls = calls.filter((x) => x.possibleAccounts.length === possibleAccountsLength);
+        if (matchingCalls.length === 0) {
+            return {
+                pass: false,
+                message: () =>
+                    `The method requestAccountSelection was called, but not with the specified possible accounts length '${possibleAccountsLength}', instead with lengths '${calls.map((e) => e.possibleAccounts.length).join(", ")}}'.`
```

Comment:
As far as I see it, there is one closing curly bracket too many at the end of the message.

Also, `${calls.map((e) => 'e.possibleAccounts.length').join(", ")}` instead of `'${calls.map((e) => e.possibleAccounts.length).join(", ")}'` should be more precise.

Reply:
Perhaps 

```
`'${e.possibleAccounts.length}'` 
```

is needed inside the mapping. 🕵️‍♀️

Reply:
your suggestion doesn't work. I hope I fixed it.

Reply:
Have you tried `${calls.map((e) => `'${e.possibleAccounts.length}'`).join(", ")}`?

Reply:
I Actually don't want this. the joined stuff should be in 's b/c it belongs together.

Reply:
Please do not confuse ' as String quotation marks. 's inside error strings make sure that the reader has a logical concatination on what's possible.